### DISCUSSION TITLE
Can I "override" a remote url?

### DISCUSSION THREAD

**Discussion by pdeluca93** [mockoon/mockoon#482] [Help] [answered]
Hi,  for example, if I have an external url that is http://10.10.10.16:8080/api/v1/test, can I override the response? 

I tried entering **8080** on port,  **api/v1** on prefix and on the route **test** but it doesn't work.

Basically, what I need is that instead of going out to reach that endpoint, intercept it on mockoon.

Thanks a lot.

**Comment by 255kb** [comment, marked as answer]
Hi @pdeluca93 I am not sure I completely understand what you are trying to achieve. Is 10.10.10.16 the IP of your computer on the network? 

One typical setup with Mockoon is to use it as a proxy. Mockoon will listen on all local network adapters (the running mock should be available on your local network from outside, for example I can call `192.168.x.x:3000/test` from another computer), and you can use the [proxy mode](https://mockoon.com/docs/latest/proxy-mode/) to forward all entering requests to a remote server. Any route that you define in Mockoon will be intercepted.

**Reply by pdeluca93** [reply]
Hi @255kb , sorry for my late response. What I want to achieve is the following scenario:

Suppose that I am developing on my pc a front-end project, that interacts with a back-end that is on a server (not on my pc), for example on the ip:port 10.10.10.16:9999.

Now, what I need is to override a backend response, for example the response of 10.10.10.16:9999/test, with Mockoon, to quickly test how my front end will work with a certain modification on the response, and I dont want to modify my front-end code to hit the "overrided" Mockoon URL. 

So, in short what I need to do is, when is needed, turn on Mockoon, and intercept that request to, for example, 10.10.10.16:9999/test by Mockoon, and return what I want to. I dont know if it is possible to achieve in any way.

Thanks a lot for your time!

**Reply by 255kb** [reply]
ok I get it now 😃 
Mockoon do not support this feature. It was requested in the past https://github.com/mockoon/mockoon/issues/67 but didn't gathered a lot of interest. 
You can achieve this by modifying your `hosts` file and setting up a redirection (probably optional).
There is a great explanation there: https://superuser.com/a/1124578
The idea is to add an entry to your hosts file: 
`127.0.0.1 10.10.10.16`
Calls to `10.10.10.16:9999` should then get redirected to `localhost:9999` (you cannot specify a destination port in the hosts file). You may achieve this a little bit differently by following the guide above, and setting up a traffic redirection from port 80 to 9999 using `netsh` on Windows, `pf` on macOS or `iptables` on Linux.

**Reply by pdeluca93** [reply]
Thanks a lot, and also thanks for founding Mockoon, its very useful in my daily life.

Greetings from Argentina.
